pkg/backup/item_backupper.go

```diff
@@ -106,35 +107,21 @@ func (ib *itemBackupper) backupItem(logger logrus.FieldLogger, obj runtime.Unstr
 	return true, []FileForArchive{}, nil
 }
 
-func (ib *itemBackupper) backupItemInternal(logger logrus.FieldLogger, obj runtime.Unstructured, groupResource schema.GroupResource, preferredGVR schema.GroupVersionResource, mustInclude, finalize bool) (bool, []FileForArchive, error) {
-	var itemFiles []FileForArchive
-	metadata, err := meta.Accessor(obj)
-	if err != nil {
-		return false, itemFiles, err
-	}
-
-	namespace := metadata.GetNamespace()
-	name := metadata.GetName()
-
-	log := logger.WithFields(map[string]interface{}{
-		"name":      name,
-		"resource":  groupResource.String(),
-		"namespace": namespace,
-	})
-
+func (ib *itemBackupper) itemInclusionChecks(log logrus.FieldLogger, mustInclude bool, metadata metav1.Object, obj runtime.Unstructured, groupResource schema.GroupResource) bool {
 	if mustInclude {
 		log.Infof("Skipping the exclusion checks for this resource")
 	} else {
 		if metadata.GetLabels()[velerov1api.ExcludeFromBackupLabel] == "true" {
```

Comment:
It's better to check the GroupResource here.
Only the PVC and PV need to go through the PV skip tracking logic.
I know it's legacy code, but the change could save some condition checks.

Reply:
Hmm. So the tracking logic is only needed for PV/PC, but the exclusion checks overall apply to all types. i.e. any item is excludable, but the skipped PV tracking is only relevant for PVC/PV -- but that func already checks inside it. ExcludeFromBackupLabel, excluded namespace or resource type, etc. All of those checks are needed for even non-PV/PVC resources.